PopApp/websockets/WebsocketUtils.js

```diff
@@ -0,0 +1,114 @@
+import { decodeUTF8, encodeBase64, decodeBase64 } from 'tweetnacl-util';
+import { sign } from 'tweetnacl';
+import { getStore } from '../Store/configureStore';
+
+/* global btoa, atob */ // do not touch! EsLint required comment!
+/* eslint-disable no-underscore-dangle */
+
+/** JSON rpc version used by our protocol */
+export const JSON_RPC_VERSION = '2.0';
+/** Number of fields a server answer exactly contains */
+export const SERVER_ANSWER_FIELD_COUNT = 3;
+
+/** Enumeration of all possible "method" fields in JsonMessages */
+export const methods = Object.freeze({
+  SUBSCRIBE: 'subscribe',
+  UNSUBSCRIBE: 'unsubscribe',
+  MESSAGE: 'message',
+  CATCHUP: 'catchup',
+  PUBLISH: 'publish',
+});
+
+/** Enumeration of all possible "object" fields in JsonMessages */
+export const objects = Object.freeze({
+  LAO: 'lao',
+  MESSAGE: 'message',
+  MEETING: 'meeting',
+});
+
+/** Enumeration of all possible "action" fields in JsonMessages */
+export const actions = Object.freeze({
+  CREATE: 'create',
+  UPDATE_PROPERTIES: 'update_properties',
+  STATE: 'state',
+  WITNESS: 'witness',
+});
+
+/** Set a new key pair for the client in the local storage */
+const _createKeyPair = () => {
+  const pair = sign.keyPair();
+  const keys = { pubKey: encodeBase64(pair.publicKey), secKey: encodeBase64(pair.secretKey) };
+  getStore().dispatch({ type: 'SET_KEYPAIR', value: keys });
+
+  return keys;
+};
+
+/** Return the user public key (base64 string) or create it if missing */
+export const getPublicKey = () => {
+  const { pubKey } = getStore().getState().keypairReducer;
+
+  // create a new keypair for the user
+  if (pubKey.length === 0) return _createKeyPair().pubKey;
+  return pubKey;
+};
+
+/** Return the user secret key (base64 string) or create it if missing */ // TODO how to do better?
```

Comment:
What do you mean by "How to do better?"

Reply:
The secret key is for now stored with the public key (no additional protection). I haven't found yet a place suitable to store securely a secret key